pgmpy/models/LinearGaussianBayesianNetwork.py

```diff
@@ -409,15 +432,21 @@ def simulate(
 
                 model.remove_node(var)
 
-        else:
-            model = self
+        # Step 3: If virtual_interventions are specified, change the CPD's of intervened variables
+        # to specified ones and remove the incoming nodes
+        if virtual_intervention != []:
```

Comment:
Don't really need this check. Right? If it is an empty list, the code block would just get skipped.